### PR TITLE
Updates fuel port sprites by retlaw34

### PR DESCRIPTION
**Pull request by quardbreak**
## Description of changes

Adds new fuel port sprites which made by retlaw34.
Now main fuel port type don't spawn tank by default (this does a hydrogen subtype).

## Why and what will this PR improve

New sprite with better quality.
More flexible tank spawn definition.

## Authorship

Sprites by retlaw34.
Minor code tweak by me.

## Changelog

:cl:retlaw34
imageadd: Added new fuel port sprites.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/fuel_port.dm [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		if(tank)
+			user.put_in_hands(tank)
+
+	update_icon()
+
+/obj/structure/fuel_port/on_update_icon()
+	cut_overlays()
+
+	if(open)
+		add_overlay(mutable_appearance(icon, "[icon_state]_open"))
```
Comment (afterthought2) [inline code/game/objects/structures/fuel_port.dm]: Why are you making these mutable appearances? You can just add the states, no?

**Review by afterthought2 — CHANGES_REQUESTED** [review]
You have to repath every instance of this used in maps.

File: code/game/objects/structures/fuel_port.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/objects/structures/fuel_port.dm]: Other than the one that sets color yep.

File: code/game/objects/structures/fuel_port.dm [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		if(tank)
+			if(tank.color)
+				add_overlay(mutable_appearance(icon, "[icon_state]_tank", tank.color))
+			else
+				add_overlay(mutable_appearance(icon, "[icon_state]_tank_orange"))
+	else
+		add_overlay(mutable_appearance(icon, "[icon_state]_closed"))
+
+/obj/structure/fuel_port/attackby(obj/item/W, mob/user)
+	if(isCrowbar(W))
```
Comment (MistakeNot4892) [inline code/game/objects/structures/fuel_port.dm]: Please use the new tool interaction proc if you're adding new code that uses tools.

**Review by MistakeNot4892 — DISMISSED** [review]

File: code/game/objects/structures/fuel_port.dm [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		if(tank)
+			user.put_in_hands(tank)
+
+	update_icon()
+
+/obj/structure/fuel_port/on_update_icon()
+	cut_overlays()
+
+	if(open)
+		add_overlay(image(icon, "[icon_state]_open"))
```
Comment (afterthought2) [inline code/game/objects/structures/fuel_port.dm]: Why `image`? Just add the state.

**Review by afterthought2 — CHANGES_REQUESTED** [review]
You didn't actually address my change request: `You have to repath every instance of this used in maps.`

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**